system/jlib/jsocket.cpp

```diff
@@ -24,6 +24,23 @@
     look at loopback
 */
 
+
+/* ISocket implementation/semantic details:
+ *
+ * 1. All sockets are created in non-blocking state. All implementations should cope with non-blocking, i.e. wait/retry.
+ * 2. read/readtms will block until min_size bytes are read or timeout expires (see readtms for more detail).
+ * read/readtms can be called with min_size=0, in which case it will return as soon as no more data is available.
+ * 3. Historically, SSL calls + non-SSL write/writetms, used to flip blocking state during the call, this was prone to error,
+ * because socket could [legitimately] be written to whilst being read from, e.g. a server could be writing to a client, and
+ * at the same time, be notified of traffic and start reading it. Flipping the blocking state of a socket is inherently unsafe
+ * in many usage patterns.
+ *
+ * The non-blocking semantics are designed so that the select handlers can read as much data as possible without blocking.
+ * e.g. dafilesrv and MP avoid blocking in their select handlers, thus avoiding up other client traffic.
```

Comment:
avoiding **holding** up ...

Reply:
>@mckellyln - please review 1st commit only. It is currently based on the pending socket refactoring PR [https://github.com/https://github.com/hpcc-systems/HPCC-Platform/pull/18841]

that was part of the jsocket changed not part of this PR. Now merged. I've rebased this PR.
Not worth going issuing a PR to fix the typo' now.